File: 1. Two Sum/solution.go
func twoSum(nums []int, target int) []int {
	numsMap := map[int]int{}
	for leftIdx := range len(nums) {
		left := nums[leftIdx]
		diff := target - left

		if rightIdx, ok := numsMap[diff]; ok {
			return []int{leftIdx, rightIdx}
		}

		numsMap[left] = leftIdx
	}

	return nil
}

File: 1. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const numsMap = new Map();
    for (let leftIdx = 0; leftIdx < nums.length; leftIdx++) {
        left = nums[leftIdx];
        const diff = target - left;
        const rightIdx = numsMap.get(diff);
        if (rightIdx !== undefined) {
            return [leftIdx, rightIdx];
        }
        numsMap.set(left, leftIdx);
    }
};
